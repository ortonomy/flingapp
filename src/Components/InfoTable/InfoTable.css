.InfoTable  {
  width: 100%;
}

.InfoTable table {
  width: 100%;
  border-collapse: collapse;
}

.InfoTable .info-title {
  padding-left: 1em;
  font-size: 1.3em;
  font-weight: bold;
  text-transform: uppercase
}

.InfoTable .info-table {
  padding: 1em 3em;
}

.InfoTable .info-table tr {
  padding: 1em 3em;
  width: 100%;
  border-bottom: solid 1px gray;
}

.InfoTable .info-table tr td {
  width: 10em;
}

.InfoTable .info-table .table-key {
    text-transform: uppercase;
}
