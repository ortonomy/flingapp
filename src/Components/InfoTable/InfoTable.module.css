.InfoTable  {
  width: 90%;
}

.header {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-between;
  align-items: center;
}

.title {
  font-size: 1.3em;
  font-weight: bold;
  text-transform: uppercase;
  padding: 2em 0;
}

.row {
  padding: 10px 0;
  margin-left: 2em;
  border-bottom: solid 1px #cccccc;
}

.cell {
  padding-left: 0px;
}

.key {
  composes: cell;
  text-transform: uppercase;
}

.edit {
  font-size:0.9em;
	font-weight: bold;
	text-align: center;
	color: #4990e2;
}
