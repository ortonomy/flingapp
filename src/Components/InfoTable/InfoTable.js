import React from 'react';
import './InfoTable.css';

const InfoTable = (props) => {
  const tableData = []
  for (let [key,value] of Object.entries(props.info)) {
    const row = <tr><td className="table-key">{key}</td><td className="table-value">{value}</td></tr>
    console.log(row)
    tableData.push(row)
  }
  return (
    <div className='InfoTable'>
      <div className='info-title'>{props.title}</div>
      <table className='info-table'>
        {tableData}
      </table>
    </div>
  )
}

export default InfoTable;
