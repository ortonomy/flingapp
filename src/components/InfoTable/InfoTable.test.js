import React from 'react';
import InfoTable from '../InfoTable';

it('renders without crashing', () => {
  const wrapper = shallow(<InfoTable />);
  expect(wrapper).toMatchSnapshot();
});
