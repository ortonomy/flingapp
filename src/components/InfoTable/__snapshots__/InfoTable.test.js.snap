// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InfoTable />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
>
          <div
                    className={undefined}
          />
          <div
                    className={undefined}
          >
                    EDIT DETAILS
          </div>
</div>,
        <Grid
          bsClass="container"
          componentClass="div"
          fluid={false}
>
          
</Grid>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className={undefined}
/>,
            <div
              className={undefined}
>
              EDIT DETAILS
</div>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "EDIT DETAILS",
              "className": undefined,
            },
            "ref": null,
            "rendered": "EDIT DETAILS",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "container",
          "children": Array [],
          "componentClass": "div",
          "fluid": false,
        },
        "ref": null,
        "rendered": Array [],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
>
            <div
                        className={undefined}
            />
            <div
                        className={undefined}
            >
                        EDIT DETAILS
            </div>
</div>,
          <Grid
            bsClass="container"
            componentClass="div"
            fluid={false}
>
            
</Grid>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className={undefined}
/>,
              <div
                className={undefined}
>
                EDIT DETAILS
</div>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "EDIT DETAILS",
                "className": undefined,
              },
              "ref": null,
              "rendered": "EDIT DETAILS",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "container",
            "children": Array [],
            "componentClass": "div",
            "fluid": false,
          },
          "ref": null,
          "rendered": Array [],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
